autoload/neomake/makers/ft/d.vim

```diff
@@ -0,0 +1,55 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#ft#d#EnabledMakers()
+    "dmd, ldmd, and gdmd all share a common CLI.
+    "Ordered in efficiency of compiler
+    let l:makers = []
+    if neomake#utils#Exists('dmd')
```

Comment:
You don't have to do exists checks yourself. Neomake checks that the program exists before it starts the job. This function could just be: `return ['dmd', 'ldmd', 'gdmd']`